src/layout/rect.rs

```diff
@@ -462,6 +499,84 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Consider using rstest for the tests to reduce the amount of vertical space. Less code to read without any complexity cost. e.g.:

```rust
#[rstest]
#[case::positiveRect::new(1, 2, 3, 4), ResizeOffset::new(5, 6), Rect::new(1, 2, 8, 10)]
#[case::negative(Rect::new(1, 2, 3, 4), ResizeOffset::new(-2, -3), Rect::new(1, 2, 1, 2)]
// ...
fn resize(#[case] rect: Rect, #[case] resize: ResizeOffset, #[case] expected) {
    assert_eq(rect.resize(resize), expected);
}
```

Reply:
Thanks, that was new to me